bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/handler/TadoStateDescriptionProvider.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.tado.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.thing.binding.BaseDynamicStateDescriptionProvider;
+import org.openhab.core.thing.i18n.ChannelTypeI18nLocalizationService;
+import org.openhab.core.thing.link.ItemChannelLinkRegistry;
+import org.openhab.core.thing.type.DynamicStateDescriptionProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link TadoStateDescriptionProvider} is responsible for providing a dynamic state description of channels whose
+ * capabilities are not hard coded in the API.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(service = { DynamicStateDescriptionProvider.class, TadoStateDescriptionProvider.class })
+public class TadoStateDescriptionProvider extends BaseDynamicStateDescriptionProvider {
+
+    @Activate
+    public TadoStateDescriptionProvider(final @Reference EventPublisher eventPublisher, //
```

Comment:
This and next line: Please remove unneeded comments at end of lines.

Reply:
Ok.